### PR TITLE
[MetaSchedule] Extract task weights during task extraction

### PR DESCRIPTION
**Pull request by junrushao1994**
Task weight is defined as the number of occurrence of a specific task that appears in a Relay function. During task extraction, we use structural equality check to determine the weights.

Also, I noted the numbering of each task could differ between task extraction and scheduled task injection, and I attribute it to different visiting order - I'm not 100% sure but it seems to work. @masahi let's revisit it later after your vacation.

CC: @zxybazh @comaniac

### CODE REVIEW

**Comment by zxybazh** [comment]
LGTM, this is import change to #10366, I'll update the PR accordingly.

**Review by zxybazh — APPROVED** [review]
LGTM, this is import change to #10366, I'll update the PR accordingly.

File: tests/python/unittest/test_meta_schedule_integration.py [hunk 1]
```diff
@@ -103,11 +103,109 @@ def test_meta_schedule_integration_extract_from_resnet():
...
+        ),
+        "fused_reshape_1": (
+            24,
+            [[1, 64, 768], [64, 768]],
+        ),
+    }
+    mod, params, _ = get_network(
+        name="bert_base",
+        input_shape=[1, 64],
+        cache_dir="/root/cache-workloads",
```
Comment (comaniac) [inline tests/python/unittest/test_meta_schedule_integration.py:194]: Just check. Are we now assuming tests are executed with root (in container)?

**Review by comaniac — APPROVED** [review]
LGTM

File: tests/python/unittest/test_meta_schedule_integration.py (hunk 1 above)
Comment (junrushao1994) [reply tests/python/unittest/test_meta_schedule_integration.py:194]: oh ooops, sorry it's only meant for my local experiments. Will remove

**Comment by masahi** [comment]
> Also, I noted the numbering of each task could differ between task extraction and scheduled task injection, and I attribute it to different visiting order

Yeah, I was aware of this issue. I didn't do anything at that time since I was not sure if the order of numbering matters at all, as long as they are unique. Even with this change, I don't know if we can guarantee that the order obtained by `PostOrderVisit -> std::reverse` is guaranteed to be the same as the one used by TE compiler.

**Review by masahi — APPROVED** [review]
I really like the simplicity of this change, compared to how weights are extracted in the auto scheduler... (`te_compiler_update_weights` thing)

**Pull request merged**